src/load.jl

```diff
@@ -24,7 +24,11 @@ function load(tf::TiffFile; verbose=true, mmap = false)
         nplanes += 1
     end
 
-    if mmap
+    ifd = first(ifds)
+    if mmap && iscontiguous(ifd) && getdata(CompressionType, ifd, COMPRESSION, COMPRESSION_NONE) === COMPRESSION_NONE
+        return MmappedTIFF(tf, ifds)
+    elseif lazyio || mmap
+        mmap && @warn "Compression and discontiguous planes are not supported by `mmap`, use `lazyio = true` instead" maxlog=1
```

Comment:
IIUC, this is a file-specific warning so should be enabled for every file, not just the first one.

```suggestion
        mmap && @warn "Compression and discontiguous planes are not supported by `mmap`, use `lazyio = true` instead"
```